lib/msf/java/rmi/builder.rb

```diff
@@ -0,0 +1,96 @@
+# -*- coding: binary -*-
+
+require 'rex/java/serialization'
+
+module Msf
+  module Java
+    module Rmi
+      module Builder
+        # Builds a RMI header stream
+        #
+        # @param opts [Hash{Symbol => <String, Fixnum>}]
+        # @option opts [String] :signature
+        # @option opts [Fixnum] :version
+        # @option opts [Fixnum] :protocol
+        # @return [Rex::Proto::Rmi::Model::OutputHeader]
+        def build_header(opts = {})
```

Comment:
This could use keyword arguments to simplify the method, e.g.:
```ruby
def build_header(
    signature: Rex::Proto::Rmi::Model::SIGNATURE,
    version: 2, 
    protocol: Rex::Proto::Rmi::Model::STREAM_PROTOCOL
)
  Rex::Proto::Rmi::Model::OutputHeader.new(
              signature: signature,
              version: version,
              protocol: protocol
  )
end